File: java/NetworkRouting/src/main/java/com/yogesh/routing/Station.java
/**
 * 
 */
package com.yogesh.routing;

import java.util.ArrayList;
import java.util.LinkedList;

/**
 * @author Yogesh Sharma
 * 
 * 
 */

public class Station implements Comparable<Station> {
	public final String name;
	public ArrayList<Route> neighbours;
	public LinkedList<Station> path;
	public double minDistance = Double.POSITIVE_INFINITY;
	public Station previous;

	public int compareTo(Station other) {
		return Double.compare(minDistance, other.minDistance);
	}

	public Station(String name) {
		this.name = name;
		neighbours = new ArrayList<Route>();
		path = new LinkedList<Station>();
	}

	public String toString() {
		return name;
	}
}

File: scala/src/main/scala/com/yogesh/routing/Station.scala
package com.yogesh.routing

import scala.collection.mutable.MutableList

case class Station(name: String) extends Ordered[Station] {
  var minDistance: Double = Double.MaxValue
  var neighbours: MutableList[Route] = MutableList[Route]()
  var path: MutableList[Station] = MutableList[Station]()

  import scala.math.Ordered.orderingToOrdered

  def compare(that: Station): Int = (this.minDistance, this.minDistance) compare (that.minDistance, that.minDistance)

}